SwiftReflector/SwiftXmlReflection/AssociatedTypeDeclaration.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Xml.Linq;
+
+namespace SwiftReflector.SwiftXmlReflection {
+	public class AssociatedTypeDeclaration {
+		public AssociatedTypeDeclaration ()
+		{
+			ConformingProtocols = new List<TypeSpec> ();
+		}
+
+		public string Name { get; set; }
+
+		public TypeSpec SuperClass { get; set; }
+		public TypeSpec DefaultType { get; set; }
+		public List<TypeSpec> ConformingProtocols { get; private set; }
+
+
+		public static AssociatedTypeDeclaration FromXElement (XElement elem)
+		{
+			var assocType = new AssociatedTypeDeclaration ();
+			assocType.Name = NameAttribute (elem);
+
+			var superClassElem = elem.Element ("superclass");
+			if (superClassElem != null) {
+				var superClassName = NameAttribute (superClassElem);
+				if (superClassName != null) {
+					assocType.SuperClass = TypeSpecParser.Parse (superClassName);
+				}
+			}
+			var defaultDefn = elem.Attribute ("defaulttype");
+			if (defaultDefn != null) {
+				assocType.DefaultType = TypeSpecParser.Parse ((string)defaultDefn);
+			}
+			
+			if (elem.Element ("conformingprotocols") != null) {
+				var conforming = from conform in elem.Element ("conformingprotocols").Elements ()
```

Comment:
In macios we tend to use LINQ's function syntax, which would look something like:

elem.Element ("conformingprotocols").Elements ().Select (conform => TypeSpecParser.Parse (NameAttribute (conform));